open-api/src/testFixtures/java/org/apache/iceberg/rest/RESTCatalogServer.java

```diff
@@ -0,0 +1,125 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+class RESTCatalogServer implements LifeCycle.Listener {
+  private static final Logger LOG = LoggerFactory.getLogger(RESTCatalogServer.class);
+
+  static final String REST_PORT = "rest.port";
+  static final int REST_PORT_DEFAULT = 8181;
+
+  private Server httpServer;
+
+  RESTCatalogServer() {}
+
+  static class CatalogContext {
+    private final Catalog catalog;
+    private final Map<String, String> configuration;
+
+    CatalogContext(Catalog catalog, Map<String, String> configuration) {
+      this.catalog = catalog;
+      this.configuration = configuration;
+    }
+
+    public Catalog getCatalog() {
+      return catalog;
+    }
+
+    public Map<String, String> getConfiguration() {
+      return configuration;
+    }
+  }
+
+  private CatalogContext initializeBackendCatalog() throws IOException {
+    // Translate environment variables to catalog properties
+    Map<String, String> catalogProperties = RCKUtils.environmentCatalogConfig();
+
+    // Fallback to a JDBCCatalog impl if one is not set
+    catalogProperties.putIfAbsent(CatalogProperties.CATALOG_IMPL, JdbcCatalog.class.getName());
+    catalogProperties.putIfAbsent(CatalogProperties.URI, "jdbc:sqlite::memory:");
+    catalogProperties.putIfAbsent("jdbc.schema-version", "V1");
+
+    // Configure a default location if one is not specified
+    String warehouseLocation = catalogProperties.get(CatalogProperties.WAREHOUSE_LOCATION);
+
+    if (warehouseLocation == null) {
+      File tmp = java.nio.file.Files.createTempDirectory("iceberg_warehouse").toFile();
+      tmp.deleteOnExit();
+      warehouseLocation = tmp.toPath().resolve("iceberg_data").toFile().getAbsolutePath();
+      catalogProperties.put(CatalogProperties.WAREHOUSE_LOCATION, warehouseLocation);
+
+      LOG.info("No warehouse location set.  Defaulting to temp location: {}", warehouseLocation);
+    }
+
+    LOG.info("Creating catalog with properties: {}", catalogProperties);
+    return new CatalogContext(
+        CatalogUtil.buildIcebergCatalog("rest_backend", catalogProperties, new Configuration()),
+        catalogProperties);
+  }
+
+  public void start(boolean join) throws Exception {
+    CatalogContext catalogContext = initializeBackendCatalog();
+
+    RESTCatalogAdapter adapter = new RESTServerCatalogAdapter(catalogContext);
+    RESTCatalogServlet servlet = new RESTCatalogServlet(adapter);
+
+    ServletContextHandler context = new ServletContextHandler(ServletContextHandler.NO_SESSIONS);
+    context.setContextPath("/");
+    ServletHolder servletHolder = new ServletHolder(servlet);
+    servletHolder.setInitParameter("javax.ws.rs.Application", "ServiceListPublic");
+    context.addServlet(servletHolder, "/*");
+    context.setVirtualHosts(null);
+    context.insertHandler(new GzipHandler());
+
+    this.httpServer =
+        new Server(
+            PropertyUtil.propertyAsInt(catalogContext.configuration, REST_PORT, REST_PORT_DEFAULT));
```

Comment:
in case this causes issues in the future, we could also just pass 0 here and the server will use a random free port.

Reply:
I played around with this a little and it's difficult to pass between the server and the tests if we rely on a default port.  Also if you're running the server for testing like pyiceberg/rust, you need to know the port, so I think sticking with a well-known port is better here.  It can be overridden if necessary.